FSW/src/util/timing.cpp

```diff
@@ -0,0 +1,237 @@
+                        // start sunrise event 
+                        sunriseTimerEvent.start();
+                        scienceMode.set(SUNRISE_MODE);
+                    }
+                    break;
+
+                case SUNRISE_MODE: // gathering data for length of buffer
+
+                    // check if it is time to change sweep direction
+                    checkSweepChange(buffer, bufIdx);
+
+                    // wait until sunrise data window is complete, save buffer,
+                    // and return to standby
+                    if (sunriseTimerEvent.checkInvoked()){
+                        saveBufferEvent.invoke();
+                        scienceMode.set(STANDBY_MODE);
+                    }
+                    break;
+            
+                default: // mode not recognized, this should not happen
+                    scienceMode.set(MODE_NOT_RECOGNIZED);
+                    break;
+            }
+
+        } else { // standby until ADCS points payload at the sun
+            Serial.println("ADCS Not Ready for Science Measurements (Timing Module)");
+            scienceMode.set(STANDBY_MODE);
+        }
+
+    } else if (currentMode == STANDBY_MODE){
+        // TODO: currently the only way to leave standby is through a command,
+        //       do we want some automated way to leave for testing?
+        //       it honestly might be best to leave it as having to manually start
+        //       a data window for testing
+        
+        // execute commands  
+        commandHandling();
+
+        // TODO: memory scrubbing  
+
+    } else { // mode was not recognized, this should not happen
+        Serial.println("Payload Mode not recognized (Timing Module) - Defaulting to Safe Mode");
+        scienceMode.set(SAFE_MODE);
+    }
+}
+
+/* - - - - - - smoothBuffer - - - - - - *
+ * Usage:
+ *  Apply basic smoothing to a configurable number of the most recent buffer entries
+ *  Avoids a single noisy measurement prematurely ending a window
+ *  To change the number of indices used in smoothing, change SMOOTH_IDX_COUNT in config.hpp 
+ * 
+ * Inputs:
+ *  buffer - array of photodiode voltages
+ *  bufIdx - index of most recent photodiode measurement
+ * 
+ * Outputs:
+ *  smoothVoltage - average of the previous SMOOTH_IDX_COUNT indices of buffer
+ */
+float smoothBuffer(float buffer[], int bufIdx){
+    float smoothVoltage = 0;
+    int i, idx;
+    for (i = bufIdx; i > bufIdx - SMOOTH_IDX_COUNT; i--){ // decrement to move backwards in time
+        
+        idx = i; // I am scared to toss the iterator for this loop through wrapBufferIdx
+        
+        // ensure that index is within array bounds 
+        // (i.e. starting with i=0 would throw error if we didn't wrap the index)
+        idx = wrapBufferIdx(idx); 
+
+        smoothVoltage += buffer[idx]; // sum voltages
+    }
+    return smoothVoltage / (float)SMOOTH_IDX_COUNT; // return mean

```

Comment:
When I first saw the function name, I was like "It's just gonna be a mean", and it was just a mean. I think I'd prefer a more idiomatic name, like `voltageMean`, or `runningVoltageMean`.

Reply:
I like `runningVoltageMean` I will make that change